### PR TITLE
RUN: Unescape test output

### PR DESCRIPTION
**Pull request by mchernyavsky**
Fixes https://github.com/intellij-rust/intellij-rust/issues/5366.

### CODE REVIEW

**Comment by mchernyavsky** [comment]
@Undin Done.

File: src/test/kotlin/org/rustSlowTests/cargo/runconfig/test/CargoTestRunnerTestBase.kt [hunk 1]
```diff
@@ -25,7 +32,7 @@ abstract class CargoTestRunnerTestBase : RunConfigurationTestBase() {
             waitFor()
         }
         UIUtil.dispatchAllInvocationEvents()
-        Disposer.dispose(executionConsole)
+        tearDownCallbacks.add { Disposer.dispose(executionConsole) }
```
[the remark this answers is not in the document]
Comment (mchernyavsky) [reply src/test/kotlin/org/rustSlowTests/cargo/runconfig/test/CargoTestRunnerTestBase.kt:35]: Looks like it doesn't work.

<details>
  <summary>Exception stacktrace</summary>

```
ERROR: Already disposed: Project (name=org.rustSlowTests.cargo.runconfig.test.CargoTestNodeInfoTest.test multiline string diff, containerState=DISPOSE_COMPLETED, componentStore=/private/var/folders/95/snw4tbr15g5bj8s74f2g2n880000gn/T/unitTest__multiline_string_diff1/org.rustSlowTests.cargo.runconfig.test.CargoTestNodeInfoTest.test multiline string diff/org.rustSlowTests.cargo.runconfig.test.CargoTestNodeInfoTest.test multiline string diff.ipr) 
com.intellij.serviceContainer.AlreadyDisposedException: Already disposed: Project (name=org.rustSlowTests.cargo.runconfig.test.CargoTestNodeInfoTest.test multiline string diff, containerState=DISPOSE_COMPLETED, componentStore=/private/var/folders/95/snw4tbr15g5bj8s74f2g2n880000gn/T/unitTest__multiline_string_diff1/org.rustSlowTests.cargo.runconfig.test.CargoTestNodeInfoTest.test multiline string diff/org.rustSlowTests.cargo.runconfig.test.CargoTestNodeInfoTest.test multiline string diff.ipr) 
	at com.intellij.serviceContainer.ComponentManagerImpl.checkStateAndGetPicoContainer(ComponentManagerImpl.kt:132)
	at com.intellij.serviceContainer.ComponentManagerImpl.getComponent(ComponentManagerImpl.kt:365)
	at com.intellij.coverage.CoverageDataManager.getInstance(CoverageDataManager.java:23)
	at com.intellij.coverage.actions.TrackCoverageAction.restoreMergedCoverage(TrackCoverageAction.java:59)
	at com.intellij.coverage.actions.TrackCoverageAction.access$000(TrackCoverageAction.java:31)
	at com.intellij.coverage.actions.TrackCoverageAction$1.dispose(TrackCoverageAction.java:78)
	at com.intellij.openapi.util.ObjectNode.lambda$execute$0(ObjectNode.java:111)
	at com.intellij.openapi.util.ObjectTree.executeActionWithRecursiveGuard(ObjectTree.java:184)
	at com.intellij.openapi.util.ObjectNode.execute(ObjectNode.java:71)
	at com.intellij.openapi.util.ObjectNode.lambda$execute$0(ObjectNode.java:95)
	at com.intellij.openapi.util.ObjectTree.executeActionWithRecursiveGuard(ObjectTree.java:184)
	at com.intellij.openapi.util.ObjectNode.execute(ObjectNode.java:71)
	at com.intellij.openapi.util.ObjectNode.lambda$execute$0(ObjectNode.java:95)
	at com.intellij.openapi.util.ObjectTree.executeActionWithRecursiveGuard(ObjectTree.java:184)
	at com.intellij.openapi.util.ObjectNode.execute(ObjectNode.java:71)
	at com.intellij.openapi.util.ObjectNode.lambda$execute$0(ObjectNode.java:95)
	at com.intellij.openapi.util.ObjectTree.executeActionWithRecursiveGuard(ObjectTree.java:184)
	at com.intellij.openapi.util.ObjectNode.execute(ObjectNode.java:71)
	at com.intellij.openapi.util.ObjectTree.executeAll(ObjectTree.java:134)
	at com.intellij.openapi.util.Disposer.dispose(Disposer.java:116)
	at com.intellij.openapi.util.Disposer.dispose(Disposer.java:106)
	at com.intellij.testFramework.UsefulTestCase.disposeRootDisposable(UsefulTestCase.java:224)
	at com.intellij.testFramework.UsefulTestCase.lambda$tearDown$1(UsefulTestCase.java:197)
	at com.intellij.testFramework.RunAll.collectExceptions(RunAll.java:58)
	at com.intellij.testFramework.RunAll.run(RunAll.java:50)
	at com.intellij.testFramework.UsefulTestCase.tearDown(UsefulTestCase.java:220)
	at com.intellij.testFramework.fixtures.CodeInsightFixtureTestCase.tearDown(CodeInsightFixtureTestCase.java:55)
	at org.rust.cargo.RsWithToolchainTestBase.tearDown(RsWithToolchainTestBase.kt:75)
	at com.intellij.testFramework.UsefulTestCase.defaultRunBare(UsefulTestCase.java:394)
	at com.intellij.testFramework.EdtTestUtil$Companion$runInEdtAndWait$1.invoke(EdtTestUtil.kt:18)
	at com.intellij.testFramework.EdtTestUtil$Companion$runInEdtAndWait$1.invoke(EdtTestUtil.kt:13)
	at com.intellij.testFramework.EdtTestUtilKt$runInEdtAndWait$3.run(EdtTestUtil.kt:67)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:301)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:416)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

CompositeException (2 nested):
------------------------------
[0]: Editor of class com.intellij.openapi.editor.impl.EditorImpl hasn't been released:
[1]: Already disposed: Project (name=org.rustSlowTests.cargo.runconfig.test.CargoTestNodeInfoTest.test multiline string diff, containerState=DISPOSE_COMPLETED, componentStore=/private/var/folders/95/snw4tbr15g5bj8s74f2g2n880000gn/T/unitTest__multiline_string_diff1/org.rustSlowTests.cargo.runconfig.test.CargoTestNodeInfoTest.test multiline string diff/org.rustSlowTests.cargo.runconfig.test.CargoTestNodeInfoTest.test multiline string diff.ipr) 
------------------------------

CompositeException (2 nested):
------------------------------
[0]: com.intellij.openapi.util.TraceableDisposable$ObjectNotDisposedException: Editor of class com.intellij.openapi.editor.impl.EditorImpl hasn't been released:
[1]: com.intellij.testFramework.LoggedErrorProcessor$TestLoggerAssertionError: Already disposed: Project (name=org.rustSlowTests.cargo.runconfig.test.CargoTestNodeInfoTest.test multiline string diff, containerState=DISPOSE_COMPLETED, componentStore=/private/var/folders/95/snw4tbr15g5bj8s74f2g2n880000gn/T/unitTest__multiline_string_diff1/org.rustSlowTests.cargo.runconfig.test.CargoTestNodeInfoTest.test multiline string diff/org.rustSlowTests.cargo.runconfig.test.CargoTestNodeInfoTest.test multiline string diff.ipr) 
------------------------------

	at com.intellij.util.lang.CompoundRuntimeException.throwIfNotEmpty(CompoundRuntimeException.java:109)
	at com.intellij.testFramework.RunAll.run(RunAll.java:50)
	at com.intellij.testFramework.UsefulTestCase.tearDown(UsefulTestCase.java:220)
	at com.intellij.testFramework.fixtures.CodeInsightFixtureTestCase.tearDown(CodeInsightFixtureTestCase.java:55)
	at org.rust.cargo.RsWithToolchainTestBase.tearDown(RsWithToolchainTestBase.kt:75)
	at com.intellij.testFramework.UsefulTestCase.defaultRunBare(UsefulTestCase.java:394)
	at com.intellij.testFramework.EdtTestUtil$Companion$runInEdtAndWait$1.invoke(EdtTestUtil.kt:18)
	at com.intellij.testFramework.EdtTestUtil$Companion$runInEdtAndWait$1.invoke(EdtTestUtil.kt:13)
	at com.intellij.testFramework.EdtTestUtilKt$runInEdtAndWait$3.run(EdtTestUtil.kt:67)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:301)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:416)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: com.intellij.openapi.util.TraceableDisposable$ObjectNotDisposedException: Editor of class com.intellij.openapi.editor.impl.EditorImpl hasn't been released:
	at com.intellij.openapi.util.TraceableDisposable.throwObjectNotDisposedError(TraceableDisposable.java:50)
	at com.intellij.openapi.editor.impl.EditorImpl.throwEditorNotDisposedError(EditorImpl.java:1021)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.throwNotReleasedError(EditorFactoryImpl.java:101)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl.validateEditorsAreReleased(EditorFactoryImpl.java:89)
	at com.intellij.openapi.editor.impl.EditorFactoryImpl$1.lambda$null$0(EditorFactoryImpl.java:64)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:201)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:310)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:80)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:128)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:46)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:184)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:974)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:847)
	at com.intellij.ide.IdeEventQueue.lambda$null$8(IdeEventQueue.java:449)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:741)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:448)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:496)
	at sun.reflect.GeneratedMethodAccessor20.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.util.ui.UIUtil.dispatchAllInvocationEvents(UIUtil.java:1794)
	at com.intellij.psi.impl.DocumentCommitThread.waitForAllCommits(DocumentCommitThread.java:202)
	at com.intellij.testFramework.UsefulTestCase.lambda$waitForAppLeakingThreads$13(UsefulTestCase.java:1125)
	at com.intellij.testFramework.EdtTestUtil$Companion$runInEdtAndWait$1.invoke(EdtTestUtil.kt:18)
	at com.intellij.testFramework.EdtTestUtil$Companion$runInEdtAndWait$1.invoke(EdtTestUtil.kt:13)
	at com.intellij.testFramework.EdtTestUtilKt.runInEdtAndWait(EdtTestUtil.kt:44)
	at com.intellij.testFramework.EdtTestUtil$Companion.runInEdtAndWait(EdtTestUtil.kt:18)
	at com.intellij.testFramework.EdtTestUtil.runInEdtAndWait(EdtTestUtil.kt)
	at com.intellij.testFramework.UsefulTestCase.waitForAppLeakingThreads(UsefulTestCase.java:1115)
	at com.intellij.testFramework.fixtures.impl.BaseFixture.lambda$tearDown$0(BaseFixture.java:44)
	at com.intellij.testFramework.RunAll.collectExceptions(RunAll.java:58)
	at com.intellij.testFramework.RunAll.run(RunAll.java:50)
	at com.intellij.testFramework.fixtures.impl.BaseFixture.tearDown(BaseFixture.java:46)
	at com.intellij.testFramework.fixtures.impl.ModuleFixtureImpl.tearDown(ModuleFixtureImpl.java:34)
	at com.intellij.testFramework.fixtures.impl.HeavyIdeaTestFixtureImpl.lambda$tearDown$1(HeavyIdeaTestFixtureImpl.java:109)
	at com.intellij.testFramework.RunAll.collectExceptions(RunAll.java:58)
	at com.intellij.testFramework.RunAll.run(RunAll.java:50)
	at com.intellij.testFramework.RunAll.run(RunAll.java:46)
	at com.intellij.testFramework.fixtures.impl.HeavyIdeaTestFixtureImpl.tearDown(HeavyIdeaTestFixtureImpl.java:166)
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.lambda$null$33(CodeInsightTestFixtureImpl.java:1275)
	at com.intellij.testFramework.EdtTestUtil$Companion$runInEdtAndWait$1.invoke(EdtTestUtil.kt:18)
	at com.intellij.testFramework.EdtTestUtil$Companion$runInEdtAndWait$1.invoke(EdtTestUtil.kt:13)
	at com.intellij.testFramework.EdtTestUtilKt.runInEdtAndWait(EdtTestUtil.kt:44)
	at com.intellij.testFramework.EdtTestUtil$Companion.runInEdtAndWait(EdtTestUtil.kt:18)
	at com.intellij.testFramework.EdtTestUtil.runInEdtAndWait(EdtTestUtil.kt)
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.lambda$tearDown$34(CodeInsightTestFixtureImpl.java:1275)
	at com.intellij.testFramework.RunAll.collectExceptions(RunAll.java:58)
	at com.intellij.testFramework.RunAll.runAll(RunAll.java:35)
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.tearDown(CodeInsightTestFixtureImpl.java:1232)
	at com.intellij.testFramework.fixtures.CodeInsightFixtureTestCase.tearDown(CodeInsightFixtureTestCase.java:48)
	... 20 more
```
</details>

**Comment by mchernyavsky** [comment]
> In general - looks ok. But:
> 
> * new tests pass even without your changes so they don't check desired behavior
> * CI says that tests fail with rust 1.32

Fixed.

File: src/test/kotlin/org/rustSlowTests/cargo/runconfig/test/CargoTestRunnerTestBase.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Undin) [reply src/test/kotlin/org/rustSlowTests/cargo/runconfig/test/CargoTestRunnerTestBase.kt:35]: Hm, indeed. Looks like we need call dispose a little but earlier. We can try to override `org.rust.cargo.RsWithToolchainTestBase#getTestRootDisposable` in the following way:
```kotlin
override fun getTestRootDisposable(): Disposable {
    return rustupFixture.testRootDisposable
}
```

[the remark this answers is not in the document]
Comment (Undin) [reply src/test/kotlin/org/rustSlowTests/cargo/runconfig/test/CargoTestRunnerTestBase.kt:35]: Or use `project` as root disposable

[the remark this answers is not in the document]
Comment (mchernyavsky) [reply src/test/kotlin/org/rustSlowTests/cargo/runconfig/test/CargoTestRunnerTestBase.kt]: Fixed.

**Comment by mchernyavsky** [comment]
@Undin Fixed.

**Pull request merged**